core/src/cubos/core/io/glfw_window.cpp

```diff
@@ -274,6 +275,34 @@ const char* GLFWWindow::getClipboard() const
 #endif
 }
 
+bool GLFWWindow::keyPressed(Key key) const
+{
+#ifdef WITH_GLFW
+    return glfwGetKey(mHandle, cubosToGlfwKey(key)) == GLFW_PRESS;
+#else
+    UNSUPPORTED();
+#endif
+}
+
+bool GLFWWindow::keyPressed(Key key, Modifiers modifiers) const
+{
+#ifdef WITH_GLFW
+    return this->keyPressed(key) && this->modifiers() == modifiers;
```

Comment:
This is a good reason as why to use `{ "ctrl-a": [ "C-a" ] }` instead of direct window access.
Input->pressed("ctrl-a") will be true even if you then press shift afterwards, only returning to false once you release A.
Window->pressed(Key::A) will go back as soon as you press other modifiers.

Reply:
I think the solution is actually adding a proper `modifiers` getter - we just have to update it correctly.
I thought about it again and I think we should go with the bit masking - if the user could query the current modifiers they would be able to handle that part of the logic if they wish to.
It should all be clear in the docs too.

Reply:
If you want to release `C-a` when `Control` is released, then does it make sense to think in modifiers anymore? I don't think so. You are basically just considering that an action is only triggered when a group of keys is pressed.
You could make it so that instead of modifiers, you just store lists of keys to pressed: `Control+A` `A+B+C` - it could be useful to not be limited to modifiers.

Reply:
I mean, you could even implement copy paste behaviour by providing yet another input primitive - sequences of keys. E.g., Ctrl must be pressed and only then should A be pressed. I think this is out of scope.

Reply:
Opened an issue, let's merge this as is